File: cSharp/SelectionSort.cs
using System;


class program
{
    
    static void SelectionSort(int [] a)
    {
        for (int i = 0; i < a.Length; i++)
        {
            int small = i;

            for (int j = i+1; j < a.Length; j++)
            {
                if (a[small] > a[j])
                {
                    small = j;
                }
            }
            (a[i], a[small]) = (a[small], a[i]);
        }
    }

    static void PrintArr(int [] a)
    {
        Console.WriteLine(string.Join(" ", a));
    }

    static void Main(String[] args)
    {
        int[] a = {65, 26, 13, 23, 12};
        Console.WriteLine("Elementos del arreglo antes de ordenarlos: ");
        PrintArr(a);
        SelectionSort(a);
        Console.WriteLine("Elementos del arreglo despues de ordenarlos:");
        PrintArr(a);
    }
}

File: c++/SelectionSort.cpp
#include <iostream>

void SelectionSort(int a[], int tam){
    for (int i = 0; i < tam ; i++){
        int small = i;
        for (int j = i + 1; j< tam; j++){
            if (a[small] > a[j]){
                small = j;
            }
        }
        std::swap(a[i], a[small]);
    }
}

void PrintArr(int a[], int tam){
    for (int i = 0; i < tam; i++) {
        std::cout << a[i] << " ";
    }
    std::cout << "\n";   
}

int main(){
    int a[] = {65, 26, 13, 23, 12};
    int tam = sizeof(a)/sizeof(a[0]);
    std::cout<<"Elemtnos del arreglo antes de ordenarlo\n";
    PrintArr(a, tam);
    SelectionSort(a, tam);
    std::cout<<"Elemtnos del arreglo despues del ordenamiento\n";
    PrintArr(a, tam);

    return 0;
}